addons/website_sale/views/templates.xml

```diff
@@ -156,6 +153,10 @@
                 </div>
                 <div class="o_wsale_product_btn"/>
             </div>
+            <t t-set="color" t-value="td_product['ribbon']['color'] or ''"/>
+            <t t-set="text_color" t-value="td_product['ribbon']['text_color'] or ''"/>
+            <t t-set="colorClass" t-value="color if color.startswith('bg-') else None"/>
+            <span t-attf-class="o_wribbon #{colorClass}" t-attf-style="color: #{text_color}; #{not colorClass and 'background-color:' + color}" t-esc="td_product['ribbon']['name'] or ''"/>
```

Comment:
As discussed above, not robust enough to check that it starts with `bg-`. I would simply have two fields instead of one in the model: one for classes, one for css color. That way no need of check here and if we later want the option to make a ribbon bold with the bold class, it is possible (so name that field generically, not bg_class)